websites/N/Neko-sama.fr/presence.ts

```diff
@@ -0,0 +1,138 @@
+const presence = new Presence({
+	clientId: "1014298173314961481"
+}),
+//You can use this to get translated strings in their browser language
+strings = presence.getStrings({
+	play: "presence.playback.playing",
+	pause: "presence.playback.paused"
+});
+  
+/*
+function myOutsideHeavyLiftingFunction(){
+	//Grab and process all your data here
+  
+	// element grabs //
+	// api calls //
+	// variable sets //
+}
+  
+setInterval(myOutsideHeavyLiftingFunction, 10000);
+//Run the function separate from the UpdateData event every 10 seconds to get and set the variables which UpdateData picks up
+*/
+
+function timeToString(nbr: number): string{
+	let nbrCopy = nbr; 
+	let nbrString: string = "";
+	if (nbrCopy >= 3600) {
+		let quotient = Math.floor(nbrCopy/3600);
+		if (isNaN(quotient)) quotient = 0
+		let remainder = nbrCopy%3600;
+		if (quotient > 9) {
+			nbrString += quotient.toString() + ":";
+		} else {
+			nbrString += "0" + quotient.toString() + ":";
+		}
+		nbrCopy = remainder;
+	}
+	let quotient = Math.floor(nbrCopy/60);
+	if (isNaN(quotient)) quotient = 0
+	let remainder = nbrCopy%60;
+	if (quotient > 9) {
+		nbrString += quotient.toString() + ":";
+	} else {
+		nbrString += "0" + quotient.toString() + ":";
+	}
+	nbrCopy = remainder;
+	if (isNaN(nbrCopy)) nbrCopy = 0
+	if (nbrCopy > 9) {
+		nbrString += nbrCopy.toString();
+	} else {
+		nbrString += "0" + nbrCopy.toString();
+	}
+	return nbrString;
+}
+
+  
+interface Video {
+	time: number;
+	duration: number;
+	paused: boolean;
+}
+
+let video: Video = null;
+
+presence.on("iFrameData", (data : Video) => {
+	video = data;
+});
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+	  	largeImageKey: "nekosama-icon",
+		buttons: [
+				{
+					label: "Voir site",
+					url: "https://www.neko-sama.fr/"
+				}
+		]
+	};
+
+	if (document.URL.startsWith("https://www.neko-sama.fr/anime/episode/")) {
+		let episodeTitle: string = document.getElementById("watch").getElementsByTagName("h1")[0].firstChild.textContent;
+		let episodeImage: string = document.head.getElementsByTagName("meta")[13].content;
+		if (episodeImage == "https://neko-sama.fr/images/default_thumbnail.png") episodeImage = "nekosama-icon";
```

Comment:
```suggestion
		const episodeImage: string = episodeImage == "https://neko-sama.fr/images/default_thumbnail.png" ? "nekosama-icon" : document.head.getElementsByTagName("meta")[13].content;		
```

Also, I'm almost sure the image is in a meta element with the property `og:image`.
So, instead of `document.head.getElementsByTagName("meta")[13].content` you should use `document.querySelector<HTMLMetaElement>('meta[property="og:image"]').content`

Reply:
> Also, I'm almost sure the image is in a meta element with the property `og:image`. So, instead of `document.head.getElementsByTagName("meta")[13].content` you should use `document.querySelector<HTMLMetaElement>('meta[property="og:image"]').content`

`document.querySelector<HTMLMetaElement>('meta[property="og:image"]').content` return "false" and I don't know much about querySelector but yes, getting name of the property is better.
I profit of that comment to thank you for your review.

`const episodeImage: string = episodeImage == "https://neko-sama.fr/images/default_thumbnail.png" ? "nekosama-icon" : document.head.getElementsByTagName("meta")[13].content;`

There, episodeImage is mention before assignment so I made that

`		const episodeImage: string = document.head.getElementsByTagName("meta")[13].content == "https://neko-sama.fr/images/default_thumbnail.png" ? "nekosama-icon" : document.head.getElementsByTagName("meta")[13].content;
`

Reply:
Yeah sorry, I've been too fast.

Reply:
No problem, I try your querySelector method and it works.